loopchain/blockchain/blockchain.py

```diff
@@ -71,19 +74,19 @@ def __init__(self, channel_name=None, store_id=None):
         self.last_unconfirmed_block = None
         self.__channel_name = channel_name
         self.__peer_id = ChannelProperty().peer_id
+        self.__block_manager: BlockManager = block_manager
 
         store_id = f"{store_id}_{channel_name}"
-        # block db has [ block_hash - block | block_height - block_hash | BlockChain.LAST_BLOCK_KEY - block_hash ]
         self._blockchain_store, self._blockchain_store_path = utils.init_default_key_value_store(store_id)
 
-        # made block count as a leader
-        self.__invoke_results = {}
+        # tx receipts and next prep after invoke, {block_hash: (receipts, next_prep)}
+        self.__invoke_results: AgingCache = AgingCache(max_age_seconds=conf.INVOKE_RESULT_AGING_SECONDS)
 
         self.__add_block_lock = threading.RLock()
         self.__confirmed_block_lock = threading.RLock()
 
         self.__total_tx = 0
-        self.__nid: str = None
+        self.__nid: Optional[str] = None
```

Comment:
Should we use Optional since now when variable can be None?

Reply:
I'm not sure. I just hated the IDE warning. How's your  opinion?

Reply:
I prefer to use `Optional` in this case.

Reply:
OK. I'm agree with you.